fabric-generic-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/generic/v1/item/ItemTooltipCallback.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.event.generic.v1.item;
+
+import java.util.List;
+
+import net.minecraft.client.item.TooltipContext;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.Text;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+@Environment(EnvType.CLIENT)
+public interface ItemTooltipCallback {
+	/**
+	 * Fired after the game has appended all base tooltip lines to the list.
+	 */
+	Event<ItemTooltipCallback> EVENT = EventFactory.createArrayBacked(ItemTooltipCallback.class, callbacks -> (stack, context, lines) -> {
```

Comment:
Should we still have this in its own separate `EVENT` class, since it looks like we're going in the direction of having classes that define lots of different events?

Reply:
Then again, a majority of the events in the API still follow that `EVENT` rule. Due to the fact there are references to client types, I couldn't easily do a `EntityUnloadCallback.CLIENT/SERVER` without classloader issues. The other route was double the amount of interfaces in the api

Reply:
Imo for that you can do `EntityUnloadCallback` and `ClientEvents.ENTITY_UNLOAD` where
`EntityUnloadCallback ENTITY_UNLOAD = EventFactory.createArrayBacked(xxx);` in `ClientEvents` class

Reply:
Also how is this a "generic event"?

Reply:
Yes, it is not really a generic event. But the item tooltip stuff is surely not lifecycle by any metric.

I'd say it makes more sense in an `item` module but this is temporary